src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -413,6 +420,28 @@ def connection_is_activated(self, connection_id):
 
         return connection_is_activated
 
+    def get_connection_for_interface(self, interface_name):
+        """Find and return the connection id currently applied to the given interface.
+
+        interface_name: The system interface name, for example eth1 or ens0s1.
+        """
+        connection_id = None
+        devices = []
+
+        try:
+            devices = self.NetworkManager.NetworkManager.GetDevices()
+        except Exception as exception:
+            self.logger.error(
```

Comment:
This should probably have those TODOs about using exception chaining once we move to Python 3.